ssc/cmod_battery.cpp

```diff
@@ -263,123 +264,125 @@ var_info vtab_battery_outputs[] = {
 
 var_info_invalid };
 
-battstor::battstor(compute_module &cm, bool setup_model, size_t nrec, double dt_hr, batt_variables *batt_vars_in)
+battstor::battstor(var_table& vt, bool setup_model, size_t nrec, double dt_hr, const std::shared_ptr<batt_variables> batt_vars_in)
 {
 	make_vars = false;
 
-	// time quantities
-	nyears = 1;
-	_dt_hour = dt_hr;
-	step_per_hour = static_cast<size_t>(1. / _dt_hour);
+    // time quantities
+    _dt_hour = dt_hr;
+    step_per_hour = static_cast<size_t>(1. / _dt_hour);
 	initialize_time(0, 0, 0);
 
-	if (cm.is_assigned("fuelcell_power")) {
-		fuelcellPower = cm.as_vector_double("fuelcell_power");
+	bool has_fuelcell = false;
+	if (auto vd = vt.lookup("fuelcell_power")) {
+		fuelcellPower = vd->arr_vector();
+		has_fuelcell = true;
 	}
 
 	// battery variables
 	if (batt_vars_in == 0) 
 	{
 		make_vars = true;
-		batt_vars = new batt_variables();
+		batt_vars = std::make_shared<batt_variables>();
```

Comment:
Please make sure all std::make_shared modifications are compliant with CentOS7

Reply:
tests pass on centos7. make_shared is c++11